src/main/java/picard/util/LiftoverUtils.java

```diff
@@ -0,0 +1,255 @@
+        }
+
+        return alleles;
+    }
+
+    private static Allele reverseComplement(Allele oldAllele, Interval target, ReferenceSequence referenceSequence, boolean isBiAllelicIndel, boolean addToStart){
+
+        if (oldAllele.isSymbolic()){
+            return oldAllele;
+        }
+        else if (isBiAllelicIndel){
+            // target.getStart is 1-based, reference bases are 0-based
+            final StringBuilder alleleBuilder = new StringBuilder(target.getEnd() - target.getStart() + 1);
+
+            if (addToStart) alleleBuilder.append((char) referenceSequence.getBases()[target.getStart() - 2]);
+            alleleBuilder.append(SequenceUtil.reverseComplement(oldAllele.getBaseString().substring(1, oldAllele.length())));
+            if (!addToStart) alleleBuilder.append((char) referenceSequence.getBases()[target.getEnd() - 1]);
+
+            return Allele.create(alleleBuilder.toString(), oldAllele.isReference());
+        }
+        else {
+            return Allele.create(SequenceUtil.reverseComplement(oldAllele.getBaseString()), oldAllele.isReference());
+        }
+    }
+
+    protected static GenotypesContext fixGenotypes(final GenotypesContext originals, List<Allele> originalAlleles, List<Allele> newAlleles) {
+        // optimization: if nothing needs to be fixed then don't bother
+        if (originalAlleles.equals(newAlleles)) {
+            return originals;
+        }
+
+        final GenotypesContext fixedGenotypes = GenotypesContext.create(originals.size());
+        for (final Genotype genotype : originals) {
+            final List<Allele> fixedAlleles = new ArrayList<>();
+            for (final Allele allele : genotype.getAlleles()) {
+                if (allele.isSymbolic() || allele.isNoCall()){
+                    fixedAlleles.add(allele);
+                }
+                else {
+                    int idx = originalAlleles.indexOf(allele);
+                    if (idx == -1) {
+                        throw new IllegalStateException("Allele not found: " + allele.toString() + ", " + originalAlleles + "/ " + newAlleles);
+                    }
+                    fixedAlleles.add(newAlleles.get(idx));
+                }
+            }
+            fixedGenotypes.add(new GenotypeBuilder(genotype).alleles(fixedAlleles).make());
+        }
+        return fixedGenotypes;
+    }
+
+    /**
+     *    Normalizes and left aligns a {@link VariantContext}.
+     *
+     *    Based on Adrian Tan, Gonçalo R. Abecasis and Hyun Min Kang. (2015)
+     *    Unified Representation of Genetic Variants. Bioinformatics.
+     *
+     * @return a new {@link VariantContext} which represents the same variation as vc but has
+     * been normalized and left-aligned
+     */
+    protected static void leftAlignVariant(VariantContextBuilder builder, int start, int end, final List<Allele> alleles, final ReferenceSequence referenceSequence) {
```

Comment:
Several comments here...

- would it be good to return the original builder as well so that one could use in functional form?
- Please make a big note about side-effect (changing the builder) in the javaDoc.
- Please convert all parameters to final
- do you need start and end as parameters? why not pull them out of builder?
- similarly for the alleles...why not pull them out of the builder?

Reply:
does builder have these getters?  it would have been nice if it did...

i'll add those notes in javadoc.  

as to returning the builder: this is basically a private method.  it's protected only so tests can access it.  in theory maybe that might be useful, but because of the deficiencies in VariantContextBuilder's get methods, i'm not sure what this by itself gains you...

Reply:
yes...would be nice to add these getters....didn't realize they are missing...

ok. about not returning.